core/src/main/java/com/graphhopper/routing/weighting/custom/CustomWeighting.java

```diff
@@ -93,12 +93,16 @@ public static CustomWeighting create(FlagEncoder baseFlagEncoder, EncodedValueLo
                                          CustomModel customModel) {
         if (customModel == null)
             throw new IllegalStateException("CustomModel cannot be null");
-        if (customModel.getMaxSpeedFallback() != null && customModel.getMaxSpeedFallback() > baseFlagEncoder.getMaxSpeed())
-            throw new IllegalArgumentException("max_speed_fallback cannot be bigger than max_speed " + baseFlagEncoder.getMaxSpeed());
-        double maxSpeedTmp = customModel.getMaxSpeedFallback() == null ? baseFlagEncoder.getMaxSpeed() : customModel.getMaxSpeedFallback();
-        SpeedAndAccessProvider speedAndAccessProvider = ExpressionBuilder.create(customModel, lookup, baseFlagEncoder.getMaxSpeed(), maxSpeedTmp, baseFlagEncoder.getAverageSpeedEnc());
+        double globalMaxSpeed = baseFlagEncoder.getMaxSpeed();
+        if (customModel.getMaxSpeedFallback() != null) {
+            if (customModel.getMaxSpeedFallback() > globalMaxSpeed)
+                throw new IllegalArgumentException("max_speed_fallback cannot be bigger than max_speed " + globalMaxSpeed);
```

Comment:
Should we throw an exception at all or just ignore MaxSpeedFallback in that case?

Reply:
I still think we should remove maxSpeedFallback entirely and derive it from the custom model.

Reply:
Ah but in case a speed larger than the base encoder speed  is specified we should raise an error I think. Otherwise it would be hard to understand why setting a too high speed yields no effect?